### PR TITLE
Add unit tests for com.baidu.hugegraph.util.StringUtil

### PR DESCRIPTION
**Pull request by Braavos6**
I've analysed your codebase and noticed that com.baidu.hugegraph.util.StringUtil is not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important.

### CODE REVIEW

File: hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package com.baidu.hugegraph.util;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+
```
Comment (JohnLBergqvist) [inline hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java]: Remove the blank line.

File: hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        assertEquals(".a\'b\'c()", desc("a\'b\'c", new ArrayList<>()));
+    }
+
+    @Test
+    public void testEscape() {
+        assertEquals("\u000011213", escape('1', '\u0000', new String[]{"1", "2", "3"}));
+        assertEquals("a1b1c", escape('1', '\u0000', new String[]{"a", "b", "c"}));
+    }
+
+
```
Comment (JohnLBergqvist) [inline hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java]: Remove the blank line.

File: hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    @Test
+    public void testUnescape() {
+        assertEquals(1, unescape("", "", "").length);
+        assertEquals(1, unescape("foo", "bar", "baz").length);
+    }
+
+    @Test
+    public void valueOfInputNotNullNotNullOutputNotNull() {
```
Comment (JohnLBergqvist) [inline hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java]: clean up the test name.

File: hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        assertEquals(12, valueOf(Integer.class, "12"));
+        assertEquals(123, valueOf(Integer.class, "123"));
```
Comment (JohnLBergqvist) [inline hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java]: Is this necessary, if you're already testing `12`? Or does a three-digit integer do something different in the source code, in which case it's fine?

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]

File: hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java (hunk 4 above)
Comment (Braavos6) [reply hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java]: I was going for double and triple digit test. I will take it out :)

Comment (JohnLBergqvist) [reply hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java]: In general, if the second assert covers the exact same lines & branches as the first assert, i'd take it out.

**Review by JohnLBergqvist — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

**Review by sudo-jaa — APPROVED** [review]

**Review by petersear — APPROVED** [review]

**Pull request closed without merging**